### ISSUE TITLE
Touch event attributes not recognized as global

### ISSUE THREAD

**Issue by alec-mitnik** [validator/validator#2038]
### Summary

The checker reports an error for having the ontouchmove attribute on a span, which it says only supports the global attributes.  But ontouchmove _is_ a global attribute!

Changing the span to a different tag such as div or button, or using a different touch event attribute such as ontouchstart, produces a similar error.

### URL for document that reproduces the problem

n/a

### HTML/CSS/SVG snippet that reproduces the problem

```html
<span ontouchmove></span>
```

### Checker error messages

```shell
Error: Attribute ontouchmove not allowed on element span at this point.

From line 7, column 1; to line 7, column 18

d>↩<body>↩<span ontouchmove></span

Attributes for element span:
Global attributes
```

### Spec links/citations

https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Global_attributes lists ontouchmove as a global attribute.

### Steps to reproduce

n/a

### Expected behavior

n/a

### Actual behavior

n/a

**Comment by j9t** [comment]
Insofar as the attributes are concerned, this looks like an error in the MDN documentation. The HTML specification itself lists [no `ontouch*` attributes](https://html.spec.whatwg.org/multipage/indices.html#attributes-3) yet. That is, while this may be in the process of standardization, it neither looks fully specified nor fully implemented yet.

@sideshowbarker, anyone, please advise. (If we want to include these four attributes regardless, happy to take care of it.)

**Comment by pkra** [comment]
@j9t https://w3c.github.io/touch-events/ has been supported in Gecko and Blink for years but never in Webkit. It is referenced in https://dom.spec.whatwg.org/#interface-event and IIUC therefore also indirectly in https://html.spec.whatwg.org/multipage/webappapis.html#event-handler-content-attributes. So it might just be an oversight on the HTML end to not list it explicitly. 

Then again, [MDN itself warns against event handler attributes](https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Attributes#event_handler_attributes).

**Comment by sideshowbarker** [comment]
Fixed in https://github.com/validator/validator/commit/e283dd662b833ed9ebe4acbc7bd51a0b0ba333b2 and deployed:

https://validator.w3.org/nu/?showsource=yes&doc=data%3Atext%2Fhtml%3Bbase64%2CPCFET0NUWVBFIGh0bWw%2BCjxodG1sIGxhbmc9ImVuIj4KPG1ldGEgY2hhcnNldD0idXRmLTgiPgo8dGl0bGU%2BVG91Y2ggZXZlbnQgaGFuZGxlciBhdHRyaWJ1dGVzPC90aXRsZT4KPGRpdiBvbnRvdWNoc3RhcnQ9InN0YXJ0KCkiIG9udG91Y2hlbmQ9ImVuZCgpIiBvbnRvdWNobW92ZT0ibW92ZSgpIiBvbnRvdWNoY2FuY2VsPSJjYW5jZWwoKSI%2BCiAgVG91Y2ggZXZlbnRzIHRlc3QKPC9kaXY%2BCg%3D%3D

**Comment by sideshowbarker** [comment]
@j9t Yeah at https://w3c.github.io/touch-events/#extensions-to-the-globaleventhandlers-mixin there is a spec for the `ontouch*` attributes. I can’t remember, but I think we may have some other spec cases like this — I mean, cases where there’s some spec other than the HTML spec which defines event-handler content attributes. And regardless, I think in such a case, our policy for the HTML spec would be to not include them in the HTML spec itself.

All that said, given that WebKit/Safari doesn’t have support for touch events, it doesn’t like a great idea for developers to be using them. But oh well. And on top of that, I believe the features in the Pointer Events spec are meant to supersede Touch Events. And so Pointer Events are what developers should probably be using instead, if anything. But those aren’t supported in WebKit/Safari either — and are likely never to be.

But anyway, the key purpose of the checker is just to help developers catch mistakes. And if somebody’s intentionally using `ontouch*` event-handler attributes, they probably don’t consider their usage to be a mistake. And so we can do to help for that case is, just catch cases where they misspelled the attribute names. So now we have that, at least.

**Comment by sideshowbarker** [comment]
> Then again, [MDN itself warns against event handler attributes](https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Attributes#event_handler_attributes).

Yeah, there is that too. It wouldn’t be completely unreasonable to argue that we could be emitting warnings for all usage of any event-handler at all. Of course we’re not going to do that — but I just mean to point it out as another case of the main purpose of the checker being, to help developers catch unintentional mistakes. And so if someone is using event-handler content attributes all over the place in their source, that’s not something they consider a mistake. So if we still want to be able to provide them with help, we can at least just make sure they didn’t misspell any of them.
